lib/asset/AssetGenerator.js

```diff
@@ -144,6 +151,18 @@ class AssetGenerator extends Generator {
 							contentHash
 						}
 					);
+					if (assetModulePublicPath) {
+						if (typeof assetModulePublicPath === "function") {
+							assetModulePublicPath = assetModulePublicPath({
+								filename,
+								...info
+							});
+						}
+						publicPath = JSON.stringify(assetModulePublicPath);
```

Comment:
```suggestion
						publicPath = JSON.stringify(
							runtimeTemplate.compilation.getAssetPathWithInfo(
								assetModulePublicPath,
								{
									module,
									runtime,
									filename: sourceFilename,
									chunkGraph,
									contentHash
								}
							)
						);
```

Reply:
Suddenly I found that if a hook is used with filename, it may affect the user's existing behavior, and now can only determine whether filename or publicPath is being modified by whether the current path has a filename suffix.

Reply:
I don't understand what you mean with that?

Reply:
.hooks.assetPath.  Shared by filename and publicPath
If the user only wants to modify the filename
Well, the expression is not very clear and can be ignored 。。。